pkg/controller/machinehealthcheck/machinehealthcheck_controller_test.go

```diff
@@ -223,102 +115,173 @@ func TestGetNodeCondition(t *testing.T) {
 
 }
 
+// newFakeReconciler returns a new reconcile.Reconciler with a fake client
+func newFakeReconciler(initObjects ...runtime.Object) *ReconcileMachineHealthCheck {
+	fakeClient := fake.NewFakeClient(initObjects...)
+	return &ReconcileMachineHealthCheck{
+		client:    fakeClient,
+		scheme:    scheme.Scheme,
+		namespace: namespace,
+	}
+}
+
 type expectedReconcile struct {
 	result reconcile.Result
 	error  bool
 }
 
-func TestReconcile(t *testing.T) {
-	nodeHealthy := node("healthy", true)
-	nodeUnhealthy := node("unhealthy", false)
-	nodeWithNoMachineAnnotation := node("noAnnotated", true)
-	nodeWithNoMachineAnnotation.Annotations = map[string]string{}
-	nodeAnnotatedWithNoExistentMachine := node("noExistentMachine", true)
-	nodeAnnotatedWithNoExistentMachine.Annotations[machineAnnotationKey] = "noExistentMachine"
-	fakeMachine := machine("fakeMachine")
-	fakeMachine.Status = mapiv1alpha1.MachineStatus{
-		NodeRef: &corev1.ObjectReference{
-			Namespace: "",
-			Name:      "healthy",
-		},
+func testReconcile(t *testing.T, remediationWaitTime time.Duration, initObjects ...runtime.Object) {
+	// healthy node
+	nodeHealthy := maotesting.NewNode("healthy", true)
+	nodeHealthy.Annotations = map[string]string{
+		machineAnnotationKey: fmt.Sprintf("%s/%s", namespace, "machineWithNodehealthy"),
```

Comment:
Here, as well. This extends the original code. Please, put it into another file.

Reply:
done

Reply:
missing a push?  i don't see a new file

Reply:
Oh :) sorry, I meant commit, not file :-|

Reply:
@ingvagabund yep I understood you :smile: I separated it into two commits, can you please check it?